Add Cloudant database resource and data source

eiri opened the pull request:
## Summary

This PR is first part part of adoption of pull request #3042. Comparing to original PR this does some minimal tweaks to Cloudant database resource and data source based on feedback from Cloudant Infrastructure team and changes internal structure to match new IBM terraform provider project structure.

The second part of original PR that'll include replication resource will be submitted separately.

<!--- See what makes a good Pull Request at : https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/.github/CONTRIBUTING.md --->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

<!--- Thank you for keeping this note for the community --->

<!--- If your PR fully resolves and should automatically close the linked issue, use Closes. Otherwise, use Relates --->

Relates to #3042

### Output from acceptance testing:

<!--
Replace TestAccXXX with a pattern that matches the tests affected by this PR.

For more information on the `-run` flag, see the `go test` documentation at https://tip.golang.org/cmd/go/#hdr-Testing_flags.
-->
```
$ make testacc TESTARGS='-run=TestAccIBMCloudant'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccIBMCloudant -timeout 700m
?   	github.com/IBM-Cloud/terraform-provider-ibm	[no test files]
?   	github.com/IBM-Cloud/terraform-provider-ibm/ibm/acctest	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns	(cached) [no tests to run]
?   	github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex	[no test files]
?   	github.com/IBM-Cloud/terraform-provider-ibm/ibm/provider	[no test files]
[WARN] Set the environment variable IBM_APPID_TENANT_ID for testing AppID resources, AppID tests will fail if this is not set

... skipped ...

[INFO] Set the environment variable IBM_CONTAINER_CLUSTER_NAME for ibm_container_nlb_dns resource or datasource else tests will fail if this is not set correctly
=== RUN   TestAccIBMCloudantDatabaseDataSourceBasic
--- PASS: TestAccIBMCloudantDatabaseDataSourceBasic (90.88s)
=== RUN   TestAccIBMCloudantDataSource_basic
--- PASS: TestAccIBMCloudantDataSource_basic (99.37s)
=== RUN   TestAccIBMCloudantDatabaseBasic
--- PASS: TestAccIBMCloudantDatabaseBasic (102.87s)
=== RUN   TestAccIBMCloudantDatabaseAllArgs
--- PASS: TestAccIBMCloudantDatabaseAllArgs (89.98s)
=== RUN   TestAccIBMCloudant_basic
--- PASS: TestAccIBMCloudant_basic (150.07s)
=== RUN   TestAccIBMCloudant_import
--- PASS: TestAccIBMCloudant_import (74.87s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cloudant	609.051s
[WARN] Set the environment variable IBM_APPID_TENANT_ID for testing AppID resources, AppID tests will fail if this is not set

... skipped ...

[INFO] Set the environment variable IBM_CONTAINER_CLUSTER_NAME for ibm_container_nlb_dns resource or datasource else tests will fail if this is not set correctly
testing: warning: no tests to run
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc	(cached) [no tests to run]
?   	github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate	[no test files]
?   	github.com/IBM-Cloud/terraform-provider-ibm/version	[no test files]
```

hkantare commented on examples/ibm-cloudant-database/modules/config-database/versions.tf:
```diff
@@ -0,0 +1,27 @@
+****************************************************/
+
+terraform {
+  required_version = ">=0.12"
```
the required_version ">=0.13" with 12 you can't declare "required_providers"

hkantare commented on website/docs/r/cloudant_database.html.markdown:
```diff
@@ -0,0 +1,52 @@
+The following arguments are supported:
+
+* `instance_crn` - (Required, string) Path parameter to specify the cloudant instance CRN.
+* `db` - (Required, Forces new resource, string) Path parameter to specify the database name.
```
The arguments/attributes should be placed in alphabatical order..same for all docs

eiri commented:
Hi @hkantare, to tell the truth I was slightly surprised to find your comments about short attributes names and usage of `d.GetOk()`, this is all the original code by @anilkumarnagaraj and I thought you folks coordinated internally in the team how you want this code to look like.

Anyway, I renamed attributes, addressed your comments and rebased the PR against current `master`. All the tests are passing. Can you please take another look when you have time?

Tests results:
```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccIBMCloudant -timeout 700m
?   	github.com/IBM-Cloud/terraform-provider-ibm	[no test files]
...
[INFO] Set the environment variable IBM_CONTAINER_CLUSTER_NAME for ibm_container_nlb_dns resource or datasource else tests will fail if this is not set correctly
=== RUN   TestAccIBMCloudantDatabaseDataSourceBasic
--- PASS: TestAccIBMCloudantDatabaseDataSourceBasic (95.84s)
=== RUN   TestAccIBMCloudantDataSource_basic
--- PASS: TestAccIBMCloudantDataSource_basic (85.78s)
=== RUN   TestAccIBMCloudantDatabaseBasic
--- PASS: TestAccIBMCloudantDatabaseBasic (101.38s)
=== RUN   TestAccIBMCloudantDatabaseAllArgs
--- PASS: TestAccIBMCloudantDatabaseAllArgs (106.48s)
=== RUN   TestAccIBMCloudant_basic
--- PASS: TestAccIBMCloudant_basic (154.11s)
=== RUN   TestAccIBMCloudant_import
--- PASS: TestAccIBMCloudant_import (82.52s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cloudant	629.699s
[WARN] Set the environment variable IBM_APPID_TENANT_ID for testing AppID resources, AppID tests will fail if this is not set
...
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc	2.981s [no tests to run]
?   	github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate	[no test files]
?   	github.com/IBM-Cloud/terraform-provider-ibm/version	[no test files]
```
```

hkantare commented on go.mod:
```diff
@@ -3,17 +3,18 @@ module github.com/IBM-Cloud/terraform-provider-ibm
 require (
-	github.com/IBM-Cloud/bluemix-go v0.0.0-20220209072802-61d92fc63fd1
+	github.com/Bowery/prompt v0.0.0-20190916142128-fa8279994f75 // indirect
+	github.com/IBM-Cloud/bluemix-go v0.0.0-20220119131246-2af2dee48688
```
Vendoring the old verison of bluemix-go

hkantare commented on website/docs/d/cloudant_database.html.markdown:
```diff
@@ -0,0 +1,68 @@
+page_title: "IBM : cloudant_database"
+description: |-
+  Get information about cloudant_database
+subcategory: "Cloudant"
```
Already the ibm_cloudant is under "Cloud Databases" should we use the same category
<img width="418" alt="Screenshot 2022-03-15 at 5 08 15 PM" src="https://user-images.githubusercontent.com/24930306/158370052-06b21c51-be10-48f4-9ea2-a81c1458299b.png">

Should we use the same or seperate all cloudant to new category ...?

If same then use the same subcategory ""Cloud Databases"  
If new then move the existing cloudant one also under new sub Category and add a entry in /website/allowed-subcategories.txt file

hkantare commented on ibm/service/cloudant/data_source_ibm_cloudant_database_test.go:
```diff
@@ -0,0 +1,62 @@
+				is_default=true
+			}
+
+			resource "ibm_resource_instance" "cloudant_instance" {
```
Why are you using ibm_resource_instance can we use ibm_cloudant resource

hkantare commented:
Can we reduce the number of commits

eiri commented:
Hi @hkantare

Sorry about the long response's delay I was consumed by a different project.

I've addressed all your comments and rebased the PR against current `master`. Take another look, please, when you have time.

The tests are passing:

```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccIBMCloudant -timeout 700m
?   	github.com/IBM-Cloud/terraform-provider-ibm	[no test files]
...
[INFO] Set the environment variable IBM_CONTAINER_CLUSTER_NAME for ibm_container_nlb_dns resource or datasource else tests will fail if this is not set correctly
=== RUN   TestAccIBMCloudantDatabaseDataSourceBasic
--- PASS: TestAccIBMCloudantDatabaseDataSourceBasic (111.01s)
=== RUN   TestAccIBMCloudantDataSource_basic
--- PASS: TestAccIBMCloudantDataSource_basic (83.77s)
=== RUN   TestAccIBMCloudantDatabaseBasic
--- PASS: TestAccIBMCloudantDatabaseBasic (120.84s)
=== RUN   TestAccIBMCloudantDatabaseAllArgs
--- PASS: TestAccIBMCloudantDatabaseAllArgs (122.26s)
=== RUN   TestAccIBMCloudant_basic
--- PASS: TestAccIBMCloudant_basic (163.91s)
=== RUN   TestAccIBMCloudant_import
--- PASS: TestAccIBMCloudant_import (80.48s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cloudant	686.695s
[WARN] Set the environment variable IBM_APPID_TENANT_ID for testing AppID resources, AppID tests will fail if this is not set
...
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc	1.187s [no tests to run]
?   	github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate	[no test files]
?   	github.com/IBM-Cloud/terraform-provider-ibm/version	[no test files]
```

hkantare commented on website/allowed-subcategories.txt:
```diff
@@ -5,7 +5,8 @@ App Configuration
 Certificate Manager
 Classic infrastructure
-Cloud Databases
+Cloudant
```
The category you mentoined is Cloudant but in docs below its subcategory: "Cloudant Databases"

hkantare replied:
If you want to seperate cloudant from Cloud Databases then please change the subcategory in exisitng reosurce/datasource of cloudan

eiri replied:
That's all right, I'll use "Cloudant Databases" for everything

hkantare commented on website/docs/d/cloudant_database.html.markdown:
````diff
@@ -0,0 +1,68 @@
+
+```hcl
+data "ibm_cloudant_database" "cloudant_database" {
+	instance_crn  = var.instance_crn
````
instead of var.XXX can you proivde an example which uses some kind of interpolation syntax reading from reosurce or datasource

hkantare commented:
Can you please resolve the conflicts to merge the PR

eiri commented:
@hkantare I've rebased against the current `master`: my tests are passing and `detect-secrets-hook --baseline .secrets.baseline` passing as well.

I've ran `detect-secrets scan --update .secrets.baseline` to get my changes into baseline and now I can see the baseline file includes some changes around `atracker` files I haven't touched, not sure why. I see there was a merge related to that module 10 hours ago and that PR didn't include `.secrets.baseline` update, so maybe this is related. https://github.com/IBM-Cloud/terraform-provider-ibm/pull/3724/files

Let me know how to go about it.
